### PR TITLE
[AL-32] Add authorize model to user

### PR DESCRIPTION
**Pull request by pvnamGit**
## Jira link
[AL-32](https://capstone-alumni.atlassian.net/jira/software/c/projects/AL/boards/1?modal=detail&selectedIssue=AL-32)
## Describe your changes
- Added `access_level`, `access_mode` to model `user`
- Removed model `role` and `user_role`

## Checklist before requesting a review
1. [x] I have performed a self-review of my code
2. [x] Have you lint your code locally before submission?

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -33,45 +33,36 @@ model Account {
 
 model User {
   id            String    @id @default(cuid())
+  archived      Boolean   @default(false)
+  createdAt     DateTime  @default(now()) @map("created_at")
+  updatedAt     DateTime? @map("updated_at")
```
[the remark this answers is not in the document]
Comment (congtrung6391) [reply prisma/schema.prisma:38]: `updatedAt` should be @updatedat. see https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#updatedat

File: prisma/schema.prisma [hunk 2]
```diff
@@ -33,45 +33,36 @@ model Account {
...
   password      String?
   image         String?
   accounts      Account[]
-  roles         UserRole[]
-  archived      Boolean   @default(false)
-  createdAt     DateTime  @default(now()) @map("created_at")
-  updatedAt     DateTime? @map("updated_at")
+  accessLevel   AccessLevel @map("access_level") @default(ALUMNI)
   accessStatus  AccessStatus  @default(PENDING) @map("access_status")
+  accessMode    AccessMode  @default(FULL_ACCESS) @map("access_mode")
```
Comment (congtrung6391) [inline prisma/schema.prisma:48]: also why do you want to map its name to snake_case?

Comment (pvnamGit) [reply prisma/schema.prisma:48]: If not use map, the field will be a camelCase in the db

File: prisma/schema.prisma (hunk 1 above)
[the remark this answers is not in the document]
Comment (pvnamGit) [reply prisma/schema.prisma:38]: same reason why I use map

[the remark this answers is not in the document]
Comment (congtrung6391) [reply prisma/schema.prisma:38]: that's ok but please add `@updatedat`

**Pull request merged**